### PR TITLE
JWT

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: lite-api/lite_api/settings.py [hunk 1]
```diff
@@ -170,3 +195,19 @@
...
+STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+
+SITE_ID = 1
+
+# Django Allauth
+SOCIALACCOUNT_PROVIDERS = {
+    'auth0': {
+        'APP': {
+            'client_id': env.str('DJANGO_ALLAUTH_AUTH0_CLIENT_ID'),
+            'secret': env.str('DJANGO_ALLAUTH_AUTH0_CLIENT_SECRET'),
```
Comment (kerin) [inline lite-api/lite_api/settings.py:208]: Can we verify that client ID and secret are required here when using RS256 as signing algorithm? The public key for signature validation can be obtained from the JWK key set provided at `tenant.auth0.com/.well-known/jwks.json`, so unless the client secret and ID are used for purposes other than validating the JWT they shouldn't be required - see:

https://auth0.com/blog/navigating-rs256-and-jwks/
https://auth0.com/docs/tokens/json-web-tokens/validate-json-web-tokens

File: lite-api/lite_api/settings.py [hunk 2]
```diff
@@ -49,13 +48,40 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'django_extensions',
+    'django.contrib.sites',
     'rest_framework',
     'oauth2_provider',
+    'allauth',
+    'allauth.account',
+    'allauth.socialaccount',
+    'allauth.socialaccount.providers.auth0',
```
Comment (kerin) [inline lite-api/lite_api/settings.py:57]: I'm slightly suspicious of using a specific Auth0 library, as here we're just using Auth0 as an OIDC-compliant stand-in for staff-sso/great-sso - could you try using a  vendor-agnostic generic OIDC library instead, such as https://github.com/mozilla/mozilla-django-oidc ?

File: lite-internal-frontend/auth/views.py [hunk 3]
```diff
@@ -77,4 +79,17 @@ def get(self, request, *args, **kwargs):
         except Exception:
             raise Exception
 
-        return redirect(getattr(settings, "LOGIN_REDIRECT_URL", "/"))
+        return redirect(settings.LOGIN_REDIRECT_URL)
+
+
+class Auth0OAuth2Adapter(views.Auth0OAuth2Adapter):
```
Comment (kerin) [inline lite-internal-frontend/auth/views.py:85]: Is this using allauth Auth0 oauth views to handle OIDC logins, or is this actually using an oauth flow?

**Review by kerin — CHANGES_REQUESTED** [review]
A couple of questions around the python library used for OIDC, and the handling of the OIDC client ID and secret.

File: lite-api/lite_api/settings.py (hunk 2 above)
Comment (richtier) [reply lite-api/lite_api/settings.py:57]: will switch to the mozilla one :+1: 

for reference tho `allauth` has a collection of hundreds of oauth2 identity providers (facebook, twitter, linkedin, aut0, etc). used it as it was quick to set up

File: lite-internal-frontend/auth/views.py (hunk 3 above)
Comment (richtier) [reply lite-internal-frontend/auth/views.py:85]: oauth flow is being used. 

by default allauth throws away the jwt and just does another reqest to auth0 like a barbarian. here were grabbing the JWT before it's thrown away.

will no longer need this once we move to the mozilla lib

**Review by kerin — APPROVED** [review]

**Pull request merged**